awx_collection/test/awx/test_instance_group.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import pytest
+
+from awx.main.models import InstanceGroup, Instance
+from awx.main.tests.functional.conftest import kube_credential, credentialtype_kube
+
+@pytest.mark.django_db
+def test_instance_group_create(run_module, admin_user):
+    result = run_module('tower_instance_group', {
+        'name': 'foo-group',
+        'policy_instance_percentage': 34,
+        'policy_instance_minimum': 12,
+        'state': 'present'
+    }, admin_user)
+    assert not result.get('failed', False), result
+    assert result['changed']
+
+    ig = InstanceGroup.objects.get(name='foo-group')
+    assert ig.policy_instance_percentage == 34
+    assert ig.policy_instance_minimum == 12
+
+    # Create a new instance in the DB
+    new_instance = Instance.objects.create(hostname='foo.example.com')
+
+    # Set the new instance group only to the one instnace
+    result = run_module('tower_instance_group', {
+        'name': 'foo-group',
+        'instances': new_instance.hostname,
```

Comment:
This is weird, instances plural, but you give it a single string, not a list?

Reply:
I'm thinking Ansible might be converting the string into an array on the fly? I'll update this so that it comes in as an Array to better match the type.